src/tribler/core/upgrade/tests/test_upgrader.py

```diff
@@ -194,6 +196,41 @@ def _exists(db, table, column):
         assert mds.get_value('db_version') == '14'
 
 
+def test_upgrade_pony14to15(upgrader: TriblerUpgrader, channels_dir, trustchain_keypair, mds_path):
+    _copy(source_name='pony_v14.db', target=mds_path)
+
+    now = int(time.time())
+
+    mds = MetadataStore(mds_path, channels_dir, trustchain_keypair, check_tables=False)
+    with db_session:
+        mds.TorrentState(infohash=unhexlify(b'0001'), seeders=0, leechers=0, self_checked=0, has_data=0, last_check=0)
+        mds.TorrentState(infohash=unhexlify(b'0002'), seeders=1, leechers=2, self_checked=0, has_data=1,
+                         last_check=now - 1000)
+        mds.TorrentState(infohash=unhexlify(b'0003'), seeders=1, leechers=2, self_checked=1, has_data=1,
+                         last_check=now - 1000)
+        mds.TorrentState(infohash=unhexlify(b'0004'), seeders=1, leechers=2, self_checked=0, has_data=1,
+                         last_check=now + 1000)
+        mds.TorrentState(infohash=unhexlify(b'0005'), seeders=1, leechers=2, self_checked=1, has_data=1,
+                         last_check=now + 1000)
+    with db_session:
+        assert mds.db.execute('select count(*) from TorrentState').fetchone()[0] == 5
+        assert mds.db.execute('select count(*) from TorrentState where self_checked > 0').fetchone()[0] == 2
+        assert mds.db.execute(f'select count(*) from TorrentState where last_check > {now}').fetchone()[0] == 2
```

Comment:
It may be possible that an SQL query contains a subtle bug, and as a result, the query always returns the same result, `0`, even if the upgrade could not fix the problem and the table still contains undesirable rows.

By checking the same query before and after the upgrade, I can test that the upgrade is indeed changed the rows in the table, and now the same query returns a different result, while before the upgrade, it was able to find undesirable rows in the table.